Implemented Car and CarWashStation classes

kram3ko opened the pull request:
with functionality for managing cars and providing car wash services. Added methods for washing cars, calculating service cost, rating the service, and maintaining a list of cars.

kram3ko committed:
Implemented Car and CarWashStation classes with functionality for managing cars and providing car wash services. Added methods for washing cars, calculating service cost, rating the service, and maintaining a list of cars.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,70 @@
+            one_car.clean_mark = self.clean_power
+            return one_car
+
+    def serve_cars(self, car_list: list | Car) -> int:
```
The type hint for `car_list` should be `list[Car]` instead of `list | Car`. The current type hint suggests that `car_list` could be a single `Car` object or a list, which is likely not the intended usage.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,70 @@
+    def __init__(self, distance_from_city_center: float,
+                 clean_power: int, average_rating: float,
+                 count_of_ratings: int) -> None:
+        if not (1 <= distance_from_city_center <= 10):
```
The condition for `distance_from_city_center` should allow for a range from 0 to 10, not 1 to 10, as a distance of 0 might be valid (e.g., if the station is in the city center).

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,70 @@
+        if not (1 <= distance_from_city_center <= 10):
+            raise ValueError("distance_from_city_center")
+        if not (1 <= average_rating <= 5):
+            raise ValueError("average raiting")
```
The error message for `average_rating` is incomplete and should be more descriptive, such as 'Average rating must be between 1 and 5.'

mateacademy-ai-mentor reviewed (changes requested):
****

Thank you for your hard work on this project! Your implementation of the `Car` and `CarWashStation` classes is well-structured and meets most of the task requirements. However, there are a few critical issues that need to be addressed before we can approve the pull request:

1. **Error Message for `average_rating`**: The error message should be more descriptive to guide users correctly. Consider updating it to something like `'Average rating must be between 1 and 5.'` to provide clear feedback.

2. **Condition for `distance_from_city_center`**: The current condition does not allow for a distance of 0, which might be valid if the station is located in the city center. Please adjust the condition to allow a range from 0 to 10.

3. **Type Hint for `car_list`**: The type hint should be `list[Car]` instead of `list | Car`. The current hint suggests that `car_list` could be a single `Car` object or a list, which is likely not the intended usage.

Once these issues are resolved, your code will be in great shape! Keep up the great work, and I look forward to seeing the improvements. 😊

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

🔄 Fix the issues, commit the changes, and then re-request my review using the `Re-request review` button in the Reviewers tab.

kram3ko committed:
- Improved error handling by adding appropriate messages for ValueErrors.
- Updated the valid range of `istance_from_city_center` from 1-10 to 0-10.
- Refactored `serve_cars` to eliminate duplicate code by reusing `calculate_washing_price` function.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,68 @@
+        if not (0 <= distance_from_city_center <= 10):
+            raise ValueError("distance from city must be between 0 and 10")
+        if not (1 <= average_rating <= 5):
+            raise ValueError("average raiting must be between 1 and 5")
```
The error message for `average_rating` should be more descriptive. Consider updating it to `'Average rating must be between 1 and 5.'` to provide clear feedback.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,68 @@
+        income += round(float(f"{formula: .2f}"), 1)
+        return income
+
+    def serve_cars(self, car_list: list[Car]) -> int:
```
The type hint for `car_list` should be `list[Car]` instead of `list | Car` to correctly indicate that it is a list of `Car` objects.

mateacademy-ai-mentor reviewed (approved):
Great job on implementing the `Car` and `CarWashStation` classes! 🚗🧼 You've effectively addressed most of the previous feedback, and your code is well-structured. Here are a few points to consider for further improvement:

1. **Error Message for `average_rating`**: You've updated the error message to `'average raiting must be between 1 and 5'`, which is more descriptive. However, ensure the spelling is correct for clarity.

2. **Type Hint for `car_list`**: You've corrected the type hint for `car_list` in the `serve_cars` method to `list[Car]`, which is appropriate and enhances code readability.

Overall, your solution meets the task requirements and constraints. Keep up the good work! Remember to review the comments to refine your code further. Addressing these minor issues will help you avoid similar problems in the future and enhance your coding skills. Keep learning and improving! 🌟

✨ Please rate this review in the Platform's notification center. I will be grateful for your feedback! ✨

kram3ko committed:
Code refactoring.
